tests/transactionScheduler_misc_test.cdc

```diff
@@ -0,0 +1,303 @@
+import Test
+import BlockchainHelpers
+import "FlowTransactionScheduler"
+import "FlowToken"
+import "TestFlowScheduledTransactionHandler"
+
+import "scheduled_transaction_test_helpers.cdc"
+
+access(all) let transactionToFail = 6 as UInt64
+access(all) let transactionToCancel = 2 as UInt64
+
+access(all) var startingHeight: UInt64 = 0
+
+access(all) var feesBalanceBefore: UFix64 = 0.0
+access(all) var accountBalanceBefore: UFix64 = 0.0
+
+access(all)
+fun setup() {
+
+    var err = Test.deployContract(
+        name: "FlowTransactionScheduler",
+        path: "../contracts/FlowTransactionScheduler.cdc",
+        arguments: []
+    )
+    Test.expect(err, Test.beNil())
+
+    err = Test.deployContract(
+        name: "TestFlowScheduledTransactionHandler",
+        path: "../contracts/testContracts/TestFlowScheduledTransactionHandler.cdc",
+        arguments: []
+    )
+    Test.expect(err, Test.beNil())
+
+    startingHeight = getCurrentBlockHeight()
+
+}
+
+access(all) fun testScheduledTransactionCancellationLimits() {
+
+    // lower the canceled transactions limit so the test runs faster
+    setConfigDetails(
+        maximumIndividualEffort: nil,
+        slotSharedEffortLimit: nil,
+        priorityEffortReserve: nil,
+        priorityEffortLimit: nil,
+        minimumExecutionEffort: nil,
+        maxDataSizeMB: nil,
+        priorityFeeMultipliers: nil,
+        refundMultiplier: nil,
+        canceledTransactionsLimit: 10,
+        collectionEffortLimit: nil,
+        collectionTransactionsLimit: nil,
+        shouldFail: nil
+    )
+
+    let currentTime = getTimestamp()
+    var timeInFuture = currentTime + futureDelta
+
+    let numToCancel: UInt64 = 20
+    var i: UInt64 = 0
+
+    // Schedule numToCancel transactions
+    while i <= numToCancel {
+
+        // Schedule a medium transaction
+        scheduleTransaction(
+            timestamp: timeInFuture + UFix64(i),
+            fee: feeAmount,
+            effort: mediumEffort,
+            priority: mediumPriority,
+            data: testData,
+            testName: "Cancellation Limits: Scheduled \(i)",
+            failWithErr: nil
+        )
+
+        i = i + 1
+
+    }
+
+    // Cancel the second transaction
+    cancelTransaction(
+        id: 2,
+        failWithErr: nil
+    )
+
+    var id: UInt64 = 1
+
+    while id < numToCancel {
+
+        // Cancel the transactions
+        if id != 2 {
+            cancelTransaction(
+                id: id,
+                failWithErr: nil
+            )
+        }
+
+        // make sure the getStatus logic is working correctly
+        // the second transaction should be canceled
+        if id < 12 {
```

Comment:
I just assume this is the cancel transactions limit value, maybe good to make it a const